runtime/runtime/src/lib.rs

```diff
@@ -2317,6 +2355,13 @@ impl Runtime {
         metrics::CHUNK_RECORDED_SIZE_UPPER_BOUND
             .with_label_values(&[shard_id_str.as_str()])
             .observe(chunk_recorded_size_upper_bound);
+
+        // Call `on_post_state_ready` with a clone of the `state_update` if requested.
+        if let Some(on_post_state_ready) = &apply_state.on_post_state_ready {
+            let post_state = PostState { trie_update: state_update.clone_for_tx_preparation() };
+            let callback = &on_post_state_ready.callback;
+            callback(post_state);
+        }
```

Comment:
mini nit: The callback is implemented in a generic fashion whereas the implementation is specialized for tx preparation. I'd recommend picking one though it's not exactly a big deal ;)